### PR TITLE
Fix invalid SQL bug in compliance-service

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
Fix bug when fetching compliance snapshots by time interval where an invalid SQL statement could be created with `IN ()` when provider_ids or policy_ids was a defined, but empty list.

## 👌 Resolves:
Invalid SQL error :c

## 📦 Impacts:
See changesets

## ✅ PR Checklist
- Added test

### CODE REVIEW

File: packages/mds-compliance-service/repository/index.ts [hunk 1]
```diff
@@ -132,10 +132,10 @@ export const ComplianceRepository = ReadWriteRepository.Create(
             .createQueryBuilder()
             .where('compliance_as_of >= :start_time', { start_time })
             .andWhere('compliance_as_of <= :end_time', { end_time })
-          if (isDefined(provider_ids)) {
+          if (isDefined(provider_ids) && provider_ids.length > 0) {
             query.andWhere('provider_id IN (:...provider_ids)', { provider_ids })
```
Comment (nessur) [inline packages/mds-compliance-service/repository/index.ts:136]:
```suggestion
            query.andWhere('provider_id = ANY (:provider_ids)', { provider_ids })
```

File: packages/mds-compliance-service/repository/index.ts [hunk 2]
```diff
@@ -132,10 +132,10 @@ export const ComplianceRepository = ReadWriteRepository.Create(
...
           }
-          if (isDefined(policy_ids)) {
+          if (isDefined(policy_ids) && policy_ids.length > 0) {
             query.andWhere('policy_id IN (:...policy_ids)', { policy_ids })
```
Comment (nessur) [inline packages/mds-compliance-service/repository/index.ts:139]:
```suggestion
            query.andWhere('policy_id = ANY (:policy_ids)', { policy_ids })
```

**Review by nessur — CHANGES_REQUESTED** [review]
small fixes, pls!
These repository queries do all 3 flavors.

File: packages/mds-compliance-service/repository/index.ts [hunk 3]
```diff
@@ -132,10 +132,10 @@ export const ComplianceRepository = ReadWriteRepository.Create(
             .createQueryBuilder()
             .where('compliance_as_of >= :start_time', { start_time })
             .andWhere('compliance_as_of <= :end_time', { end_time })
-          if (isDefined(provider_ids)) {
+          if (isDefined(provider_ids) && provider_ids.length > 0) {
```
Comment (drtyh2o) [inline packages/mds-compliance-service/repository/index.ts:135]: One could make the argument that `= ANY([])` should return 0 rows by definition.

**Review by drtyh2o — APPROVED** [review]

File: packages/mds-compliance-service/repository/index.ts (hunk 1 above)
Comment (janedotx) [reply packages/mds-compliance-service/repository/index.ts:136]: I like this suggestion.

File: packages/mds-compliance-service/tests/index.spec.ts [hunk 4]
```diff
@@ -123,6 +123,14 @@ describe('ComplianceSnapshots Service Tests', () => {
     }
   })
 
+  it('Gets ComplianceSnapshots by TimeInterval with empty provider_ids list', async () => {
+    await expect(
+      ComplianceServiceClient.getComplianceSnapshotsByTimeInterval({
```
Comment (janedotx) [inline packages/mds-compliance-service/tests/index.spec.ts:128]: Was the previous behavior causing this execution path to throw since the SQL was invalid?

Comment (avatarneil) [reply packages/mds-compliance-service/tests/index.spec.ts:128]: Yep

File: packages/mds-compliance-service/repository/index.ts (hunk 3 above)
Comment (PeterSP) [reply packages/mds-compliance-service/repository/index.ts:135]: What does the code calling us look like that is sometimes doing this?

**Review by nessur — APPROVED** [review]

**Review by PeterSP — COMMENTED** [review]
Is this properly a bug in the platform code? It seems rather strange that the front end is sending [] instead of nothing for that param.

**Review by janedotx — APPROVED** [review]

**Pull request merged**